engine/errors.go

```diff
@@ -0,0 +1,51 @@
+package engine
+
+import "errors"
+
+var (
+	// ErrFeatureNotSupported is a common error that may be returned from optional
+	// Engine methods to indicate the engine implementation doesn't support the
+	// given feature.
+	//
+	// Note, all methods are allowed to return this error, some methods are
+	// required, and may not return this error.
+	//
+	// When the worker encounters this error from an optional method, it should
+	// workaround if possible, but most likely resolve the task as "exception"
+	// with reason "malformed-payload".
+	ErrFeatureNotSupported   = errors.New("Feature not support by current engine")
+	ErrResourceNotFound      = errors.New("The referenced resource wasn't found")
+	ErrSandboxTerminated     = errors.new("The Sandbox has terminated")
+	ErrSandboxAborted        = errors.new("Exection of sandbox was aborted")
+	ErrNonFatalInternalError = errors.new("Engine encountered a non-fatal internal error")
+	ErrContractViolation     = errors.new("Engine has detected a contract violation")
+	ErrEngineIsSingleton     = errors.New("Engine cannot run multiple sandboxes in parallel")
+	ErrEngineNotSupported    = errors.New("Engine is not available in the current configuration")
+	ErrEngineUnknown         = errors.New("Engine with the given doesn't exist")
+)
```

Comment:
I think this makes sense for some errors, like the `MalformedPayloadError` which is defined below...

But if the current engine returns `ErrFeatureNotSupported` from `AttachProxy(...)` then it's clearly evident from the context that `AttachProxy` feature isn't supported on the current engine.

The plugin that gets this error, must then either: (A) decide that the task can proceed without the proxy (perhaps injecting the data some other way), or (B) return an ErrMalformedPayload message saying that `engine.Name()` doesn't support `AttachProxy`, hence, `<my-feature>` isn't supported here... And you must formulate your task payload in a manner than doesn't active this feature.

Adding `engine.Name()` might be nice... But I also fear someone could use it to write a plugin that is engine specific.